torch/csrc/api/include/torch/nn/modules/conv.h

```diff
@@ -48,7 +42,13 @@ class TORCH_API ConvImpl : public torch::nn::Cloneable<Derived> {
 /// the exact behavior of this module.
 class TORCH_API Conv1dImpl : public ConvImpl<1, Conv1dImpl> {
  public:
-  using ConvImpl<1, Conv1dImpl>::ConvImpl;
+  Conv1dImpl(
+      int64_t input_channels,
+      int64_t output_channels,
+      ExpandingArray<1> kernel_size)
+      : Conv1dImpl(ConvOptions<1>(input_channels, output_channels, kernel_size)) {
+  }
```

Comment:
I moved the constructor that takes `input_channels, output_channels, kernel_size` into each of the Conv{1,2,3}d subclasses, so that they can call each dimension's specific constructor that takes `options` (which contains specialized logic). This mirrors the design of the Python version.